scripts/13-why-rust.md

```diff
@@ -0,0 +1,469 @@
+<li>No bytecode</li>
+<li>No JIT</li>
+<li>No Garbage Collector</li>
+
+</ul>
+
+</split>
+
+notes:
+
+Rust's compromises are EXCELLENT.
+
+The language has all the features I want, and none of the ones I don't.
+
+Rust can basically do EVERYTHING.
+
+
+---
+
+> My biggest compliment to Rust is that it's boring, and this is an amazing compliment.
+
+– Chris Dickinson, on internal npm Rust adoption
+
+notes:
+
+It's not even that interesting.
+Rust isn't trendy, it's boring.
+
+---
+
+# Here Be Kraken
+
+notes:
+
+You know what isn't boring? seafaring in the ancient Mediterranean.
+
+If you were a sailor or at merchant, you would understand that you would lose ships to the sea now and then.
+
+So it was, so it had always been.
+
+Most of the sea was unexplored, and there were monsters, no doubt, taking the ships.
+
+Sailors accepted that these were just the risks that you had to accept to operate on the high seas.
+
+---
+
+# Here Be Overheads
+
+notes:
+
+You know what else isn't boring? SOFTWARE.
+
+And I don't mean in a good way.
+
+In our field in 2022, everyone accepts that there are some things you just have to deal with in order to write software.
+
+---
+
+# Lies Programmers Believe About Software
+
+notes:
+
+some lies programmers believe about software, include, but are not limited to:
+
+- That software is built on unknowable layers of abstraction
+- The layers can only do what you want if you recite the correct inscrutable incantations
+- Scaling horizontally is the way to solve your problems
+- Bugs must be caught in production.
+- If you don't use yaml and javascript to solve your problems, you will have to use xml and java.
+- speed
+
+---
+
+## I Am Tired of Accepting Things I Cannot Change
+
+I am now changing things I cannot accept
+
+notes:
+
+I have been a web developer for 15 years, writing backends in python, ruby and node, and frontends in various javascript frameworks.
+
+I am tired of writing bullshit that I can't guarantee.
+
+and I'm not alone.
+
+---
+
+![[linus-easy-going.png]]
+
+(with apologies to Linus)
+
+notes:
+
+Linus Torvalds is not an easy man to please.
+
+As you probably know, Linus created and oversees the development of the Linux kernel.
+
+He is a detail-oriented man who cares about reliability over all else.
+
+And yet when Rust support was recently merged in to the Linux kernel, many veteran developers decried this step as bowing to a flashy, fashionable new language.
+
+The idea that Linus would approve of rust due to POPULARITY is laughable to those who know his reputation.
+
+---
+
+![[network diagram.png]]
+
+notes:
+
+And Linus is ABSOLUTELY RIGHT to be so conservative.
+
+Nearly every computer used show you this video, from your wifi router, to the network systems in your school, company and isp, to google's global video distribution network, is running this kernel.
+
+There are islands of mac devices. But Linux is the OCEAN.
+
+There are clouds of windows networks. But Linux is the SKY.
```

Comment:
Oh my god I love this.

Reply:
Then you'll really love the song that inspired these lines https://www.youtube.com/watch?v=9ZDAYg196x8